lib/Cryptol.cry

```diff
@@ -340,13 +340,20 @@ primitive (%$) : {a} (Arith a) => a -> a -> a
  * Unsigned carry.  Returns true if the unsigned addition of the given
  * bitvector arguments would result in an unsigned overflow.
  */
-primitive carry : {n} (fin n) => [n] -> [n] -> Bit
+carry : {n} (fin n) => [n] -> [n] -> Bit
+carry x y = (x + y) < x
```

Comment:
It's probably not too important, but the alternate definition `~ x < y` might be slightly simpler (i.e. produces slightly smaller symbolic formulas)
```
Cryptol> :prove \(x:[8]) y -> carry x y == (~ x < y)
Q.E.D.
```

Reply:
Interesting... 

I wonder though: the first one only involves the "arithmetic fragment" of the theory, whereas the second one mixes the bitwise and arithmetic fragment.  I have an intuition that solvers generally do better if you can keep things in the arithmetic fragment because then it doesn't have to bitblast as aggressively.

Reply:
This would be pretty easy to revisit later if we want, so I'm leaving it as-is for now.